machine/machine.go

```diff
@@ -47,26 +42,19 @@ var (
 	memoryCapacityRegexp = regexp.MustCompile(`MemTotal:\s*([0-9]+) kB`)
 	swapCapacityRegexp   = regexp.MustCompile(`SwapTotal:\s*([0-9]+) kB`)
 
-	cpuBusPath         = "/sys/bus/cpu/devices/"
-	isMemoryController = regexp.MustCompile("mc[0-9]+")
-	isDimm             = regexp.MustCompile("dimm[0-9]+")
-	machineArch        = getMachineArch()
-	maxFreqFile        = "/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq"
-)
+	cpuBusPath = "/sys/bus/cpu/devices/"
 
-const sysFsCPUCoreID = "core_id"
-const sysFsCPUPhysicalPackageID = "physical_package_id"
-const sysFsCPUTopology = "topology"
-const memTypeFileName = "dimm_mem_type"
-const sizeFileName = "size"
+	machineArch = getMachineArch()
+	maxFreqFile = "/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq"
+)
 
 // GetPhysicalCores returns number of CPU cores reading /proc/cpuinfo file or if needed information from sysfs cpu path
 func GetPhysicalCores(procInfo []byte) int {
```

Comment:
one additional question: I came across this function https://github.com/google/cadvisor/blob/master/container/libcontainer/handler.go#L813-L819 which also looks to get number of online cpus via calling sched_getaffinity(2). I'm not sure how that works with hotplug, but just curious if that would be helpful here...

Reply:
I would dare to say that the function has an unfortunate name: it returns count of CPUs that current process can run on: they are undoubtedly online, but they might be affected by cpusets or isolcpus.

Reply:
I see, got it, thanks for the info.